packages/authentication/src/context/OidcContext.tsx

```diff
@@ -0,0 +1,102 @@
+import React, { useEffect } from "react";
+import { AuthProvider } from "react-oidc-context";
+import { User } from "oidc-client-ts";
+import { useState } from "react";
+import { DecodedToken } from "../interfaces/decoded-token";
+import ProtectedApp from "../components/ProtectedApp";
+import { decodeToken } from "../utils/decode-token";
+import generateRedirectUri from "../utils/generate-redirect-uri";
+
+export type AuthenticationMethods = {
+  person?: string[];
+  company?: string[];
+  proxy?: string[];
+};
+
+export type OidcConfig = {
+  url: string;
+  clientId: string;
+  realm: string;
+  redirectUri: string;
+};
+
+export type SessionLengthManagementProps = {
+  autoIdleSessionLogout?: boolean;
+  idleTimeoutMinutes?: number;
+};
+
+export type OidcProviderProps = OidcConfig &
+  SessionLengthManagementProps & {
+    children: React.ReactNode;
+    authenticationMethods?: AuthenticationMethods;
+  };
+
+export interface OidcContextInterface {
+  oidcToken: string;
+  setOidcToken: (token: string) => void;
+  decodedToken: DecodedToken | undefined;
+  authenticationMethods?: AuthenticationMethods;
+}
+
+const OidcContext = React.createContext<OidcContextInterface>(
+  {} as OidcContextInterface,
+);
+
+export const OidcProvider = ({
+  authenticationMethods,
+  url,
+  clientId,
+  realm,
+  redirectUri,
+  children,
+  autoIdleSessionLogout,
+  idleTimeoutMinutes,
+}: OidcProviderProps) => {
+  const [oidcToken, setOidcToken] = useState("");
+  const [decodedToken, setDecodedToken] = useState<DecodedToken | undefined>(
+    undefined,
+  );
+
+  useEffect(() => {
+    if (oidcToken) {
+      setDecodedToken(decodeToken(oidcToken));
+    }
+  }, [oidcToken]);
+
+  const oidcConfig = {
+    authority: `${url}/realms/${realm}`,
+    client_id: clientId,
+    redirect_uri: generateRedirectUri(redirectUri),
+    accessTokenExpiringNotificationTime: 15,
```

Comment:
Please also add `post_logout_redirect_uri` as a prop. We don't want to be stuck on the "Successfully logged out" screen of the IDP.

Reply:
Thanks! I've added the prop from the config variable, and by default i use the redirectUri, which works like it did with the old react-keycloak dependency.